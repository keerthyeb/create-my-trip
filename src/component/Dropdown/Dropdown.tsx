import {FormControl, InputLabel, MenuItem, OutlinedInput, Select} from "@mui/material";

const Dropdown = (props: { options: any; placeHolder: string; onChangeHandler: any }) => {
    const {options, placeHolder, onChangeHandler} = props;
    return (
        <FormControl sx={{m: 0, width: 300}}>
            <InputLabel>{placeHolder}</InputLabel>

            <Select
                onChange={onChangeHandler}
                input={<OutlinedInput label={placeHolder}/>}
            >
                {
                    options.map((option: string) => <MenuItem key={option} value={option}>
                        {option}</MenuItem>)
                }

            </Select>
        </FormControl>

    );
}


export default Dropdown;
