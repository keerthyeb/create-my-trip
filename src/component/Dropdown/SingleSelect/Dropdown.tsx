import { FormControl, InputLabel, MenuItem, OutlinedInput, Select } from '@mui/material';

interface DropdownProps {
  options: string[];
  placeHolder: string;
  onChangeHandler: any;
}

const Dropdown: React.FC<DropdownProps> = ({ options, placeHolder, onChangeHandler }) => {
  return (
    <FormControl sx={{ m: 0, width: 300 }}>
      <InputLabel>{placeHolder}</InputLabel>

      <Select onChange={onChangeHandler} input={<OutlinedInput label={placeHolder} />}>
        {options.map((option: string) => (
          <MenuItem key={option} value={option}>
            {option}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default Dropdown;
