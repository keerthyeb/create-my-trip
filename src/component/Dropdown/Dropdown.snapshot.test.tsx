import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from './Dropdown';

describe('Dropdown Component', () => {
  it('should render Dropdown component correctly', () => {
    const options = ['Option 1', 'Option 2', 'Option 3'];
    const placeHolder = 'Select an option';
    const onChangeHandler = jest.fn();

    const component = renderer.create(
      <Dropdown options={options} placeHolder={placeHolder} onChangeHandler={onChangeHandler} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
