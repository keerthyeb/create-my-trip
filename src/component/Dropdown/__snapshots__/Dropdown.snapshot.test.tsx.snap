// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render Dropdown component correctly 1`] = `
<div
  className="MuiFormControl-root css-cquceo-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Select an option
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      aria-controls=":r0:"
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        â€‹
      </span>
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-yjsfm1"
      >
        <span>
          Select an option
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
